dialdbg/src/main.rs

```diff
@@ -0,0 +1,239 @@
+mod parse;
+mod strings;
+
+use anyhow::{anyhow, Result};
+use clap::Parser;
+use log4rs::append::file::FileAppender;
+use log4rs::config::{Appender, Config, Root};
+use serde_json::{from_str, Value};
+use std::{fs, io, path::PathBuf};
+use tokio::time::Instant;
+use viam::rpc::dial::{self, ViamChannel};
+use webrtc::stats;
+
+/// dialdbg gives information on how to connect to a Viam robot
+#[derive(Parser, Debug)]
+#[command(author, version, about)]
+struct Args {
+    /// Filepath for remote configuration JSON containing URI and secret
+    #[arg(short, long)]
+    config: Option<PathBuf>,
+
+    /// Whether direct gRPC connection should not be examined
+    #[arg(long, action, conflicts_with("nowebrtc"))]
+    nogrpc: bool,
+
+    /// Whether WebRTC connection should not be examined
+    #[arg(long, action, conflicts_with("nogrpc"))]
+    nowebrtc: bool,
+
+    /// Filepath for output of dialdbg (file will be overwritten)
+    #[arg(short, long)]
+    output: Option<PathBuf>,
+
+    /// Secret with which to connect to the robot
+    #[arg(short, long, required_unless_present("config"))]
+    secret: Option<String>,
+
+    /// URI of the robot to connect to
+    #[arg(short, long, required_unless_present("config"))]
+    uri: Option<String>,
+}
+
+fn read_config(config: PathBuf) -> Result<(String, String)> {
+    let (uri, secret): (String, String);
+    let r: Value = from_str(fs::read_to_string(config)?.as_str())?;
+    if let Some(u) = r.get("address") {
```

Comment:
Would rewrite this in a more fucntional way like 
`let uri = r
        .get("address")
        .and_then(|e| e.as_str())
        .and_then(|e| Some(e.to_string()))
        .ok_or(anyhow!(
            "an error"
        ))?;`

or : 
`
let uri = r
        .get("address")
        .ok_or(anyhow!("an error"))?
        .as_str()
        .ok_or(anyhow!(
            "another error"
        ))?
        .to_string();
`

Reply:
also anyhow is great but given the repetitions of errors would consider using thiserror for reusable errors

Reply:
This function no longer exists.

> also anyhow is great but given the repetitions of errors would consider using thiserror for reusable errors

Will look into this! Thanks.

Reply:
Without `read_config`, I'm not sure there's enough reused errors in `parse.rs` to warrant a stronger error system right now. Let me know if you still feel strongly.